pkg/util/resize/configspec.go

```diff
@@ -463,6 +466,23 @@ func compareVmxStatsCollectionEnabled(
 	cmpPtr(ci.VmxStatsCollectionEnabled, cs.VmxStatsCollectionEnabled, &outCS.VmxStatsCollectionEnabled)
 }
 
+// compareMemoryReservationLockedToMax compares full memory reservation settings from config spec.
+func compareMemoryReservationLockedToMax(
+	ci vimtypes.VirtualMachineConfigInfo,
+	cs vimtypes.VirtualMachineConfigSpec,
+	outCS *vimtypes.VirtualMachineConfigSpec) {
+
+	memLockedMax := cs.MemoryReservationLockedToMax
+	if memLockedMax != nil && !*memLockedMax {
+		// If the desired configSpec has PCI pass-through devices, make sure memoryReservationLockedToMax is not false
```

Comment:
After the refactor of this function, maybe reword this a little - " .... memoryReservationLockedToMax must be true because ... " since at this point we know that locked is `&false`